modules/apps/item-selector/item-selector-taglib/src/main/resources/META-INF/resources/item_selector_uploader/js/utils/sendFile.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+const noop = () => {};
+
+function parse(req) {
+	var result;
+	try {
+		result = JSON.parse(req.responseText);
+	}
+	catch (error) {
+		result = req.responseText;
+	}
+
+	return result;
+}
+
+export default function sendFile({
+	file,
+	fileFieldName,
+	onProgress = noop,
+	onSuccess = noop,
+	onError = noop,
+	url,
+}) {
+	const formData = new FormData();
+	const request = new XMLHttpRequest();
```

Comment:
Is there any reason you're using `XMLHttpRequest` instead of `fetch` here? (I'm assuming it's because of the `'progress'` event but I wasn't sure. You might want to leave a comment or something similar in the PR description to avoid confusion

Reply:
Yep, it's because of the progress. I will update the description.